### PR TITLE
Rawstring

### PR DESCRIPTION
**Pull request by Movax12**
Hi,

I think the main issue here was the documentation. If Greg can take a look, I can't remember how it was suggested to change it. The code seems to work well. 

If at any point my proposed changes don't seem to suit cc65/ca65 don't be afraid to say so.
(I think this one is good/helpful but I have some ideas down the line that maybe more radical.)

Anyone, take a look at this when you can, thanks.

### CODE REVIEW

File: doc/ca65.sgml [hunk 1]
```diff
@@ -652,7 +652,54 @@ problem in most cases.
 
 To force a specific order of evaluation, parentheses may be used, as usual.
 
+<sect>String Constants and Raw-String Constants<label id="String_Constants"><p>
```
Comment (greg-king5) [inline doc/ca65.sgml]: It should be a `<sect1>`.  It should be called `<sect1>String and Raw-String formats<p>`.  It should be put between `<sect1>Number format<p>` (line 504) and `<sect1>Conditional assembly<p>` (line 512).

File: doc/ca65.sgml [hunk 2]
```diff
@@ -854,8 +899,6 @@ cheap local labels are written to the object file and will be available in the
 symbol file via the linker. Unnamed labels are not written to the object file,
 because they don't have a name which would allow to access them.
 
-
-
 <sect>Scopes<label id="scopes"><p>
```
Comment (greg-king5) [inline doc/ca65.sgml]: Look at <https://github.com/cc65/wiki/wiki/LinuxDoc-sources-style-guide> for a guide about the line spacing between sections.

File: src/ca65/scanner.c [hunk 3]
```diff
@@ -403,29 +404,111 @@ static void IFNextChar (CharSource* S)
...
+        } else {
+
+            /* Append data to line */
+            SB_AppendChar (&S->V.File.Line, N);
+
+        }
     }
 
+    /* We didn't copy the EOL in the while loop, so add it now */
+    SB_AppendChar (&S->V.File.Line, '\n');
```
Comment (greg-king5) [inline src/ca65/scanner.c]: You should copy the EOL in the while loop.  That is, you should "collect stuff" before you test for EOL.

File: src/ca65/scanner.c [hunk 4]
```diff
@@ -774,6 +914,110 @@ static void ReadIdent (void)
...
+                    /* Scan through the keyword if the char matches */
+                    unsigned K = 1;
+                    NextChar ();
+                    while ((K < D.Len) && (C == D.Buf[K]))  {
+                        NextChar ();
+                        K++;
+                    }
+
+                    /* If was not a match, append anything we skipped to the string literal.
+                    ** If Len is less then the length of the delimiter OR the next char is NOT a quote it wasn't a match.
```
Comment (greg-king5) [inline src/ca65/scanner.c]: Typo: then -> than

File: src/ca65/scanner.c (hunk 3 above)
Comment (Movax12) [inline src/ca65/scanner.c]: I think I should change the comment to reflect better what is happening:

```C
/* Either we have found EOF, or a newline, so mark the end of the line with a newline */
```

This is very close to what the origonal code is doing.

**Comment by oliverschmidt** [comment]
I must admit that I don't have an overview on the open issues with the pull request. Hower at least the ```then``` -> ```than``` typo was not taken care of for 1/2 year by now. If there's no more momentum in this pull request I'll close it soon.

**Comment by oliverschmidt** [comment]
Any takers? Otherwise to be closed next week.

**Comment by oliverschmidt** [comment]
Gone.

**Pull request closed without merging**